### PR TITLE
Dev 1446: Add Connect API Reference

### PR DESCRIPTION
**Pull request by ShreyaThapa**

### CODE REVIEW

File: pages/api-reference/connect/authorization/index.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+meta:
+  title: API Reference | Dwolla Authorization Overview
+  description: "Obtain an OAuth access token to access data in the Dwolla API on behalf of an account or application. Grant permission with the client credentials flow."
+---
+
+import guideIcon from "../../../../assets/images/content-images/content-icons/guides-icon.svg";
+
+# Authorization
+
+Dwolla utilizes the [OAuth 2 protocol](https://oauth.net/2/) to facilitate authorization. OAuth is an authorization framework that enables a third-party application to obtain access to protected resources (Transfers, Funding Sources, Customers etc.) in the Dwolla API. Access to the Dwolla API can be granted to an application either on behalf of a user or on behalf of the application itself. This section covers application auth which is meant for server-to-server applications using the Dwolla API.
```
Comment (spencerhunter) [inline pages/api-reference/connect/authorization/index.mdx]: Edit: Replace `Customers` with External Parties

File: pages/api-reference/connect/authorization/index.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#### Creating an application
```
Comment (spencerhunter) [inline pages/api-reference/connect/authorization/index.mdx]: Let's plan on removing this section for now

File: pages/api-reference/connect/root/index.mdx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+guide:
+  step: 0
+meta:
+  title: API Reference | Root
+  description: "Provide your application with the ability to fetch and discover resources available based on the OAuth `access_token` provided in the request"
+---
+
+# Root
+
+The "root" serves as an entry point to the API, providing your application with the ability to fetch and discover resources available based on the OAuth `access_token` provided in the request. If an application access token is provided in the request, the API will return links to resources that belong to the Dwolla application (i.e. "events" and "webhook-subscriptions").
```
Comment (spencerhunter) [inline pages/api-reference/connect/root/index.mdx]: Remove parenthesis at end of paragraph.
```suggestion
The "root" serves as an entry point to the API, providing your application with the ability to fetch and discover resources available based on the OAuth `access_token` provided in the request. If an application access token is provided in the request, the API will return links to resources that belong to the Dwolla application.
```

File: pages/api-reference/connect/root/index.mdx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+### Request and response
+
+```bash
+GET https://api-sandbox.dwolla.com/
+Accept: application/vnd.dwolla.v1.hal+json
+Authorization: Bearer pBA9fVDBEyYZCEsLf/wKehyh1RTpzjUj5KzIRfDi0wKTii7DqY
+
+...
+
+{
```
Comment (spencerhunter) [inline pages/api-reference/connect/root/index.mdx:32]: Updated example to include just the account link
```suggestion
{
    "_links": {
        "account": {
            "href": "https://api-sandbox.dwolla.com/accounts/d4a0876f-642e-478b-b684-bfb8632c5937",
            "type": "application/vnd.dwolla.v1.hal+json",
            "resource-type": "account"
        }
    }
}
```

File: assets/api-resource-tables/accounts/CreateFundingSource.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+              </span>,
+            ],
+            HalLink("treasury-account"),
+          ],
+        ],
+      },
+    ],
+    [
+      "accountNumber",
+      "required",
```
Comment (spencerhunter) [inline assets/api-resource-tables/accounts/CreateFundingSource.tsx]: Can we change the required value from required to yes or no?
```suggestion
      "yes",
```

File: pages/api-reference/connect/transfers/initiate.mdx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+| HTTP Status | Error Message             | Description                                                                                    |
+| ----------- | ------------------------- | ---------------------------------------------------------------------------------------------- |
+| 201         | Created.                  | A transfer was created.                                                                        |
+| 400         | Funding source not found. | Double check the funding source Id, and make sure you are using the correct funding source Id. |
+| 400         | Invalid amount            | The supplied amount must be a positive number.                                                 |
+| 401         | Invalid access token      | Access token not valid. Generate a new one and try again.                                      |
+| 403         | Forbidden                 | Not authorized to create a transfer.                                                           |
+
+## Request and response - using `ach`
```
Comment (spencerhunter) [inline pages/api-reference/connect/transfers/initiate.mdx]: Let's remove `- using ach` for now until we have more transfer types to demonstrate in the docs

File: pages/api-reference/connect/webhooks/index.mdx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+meta:
+  title: Webhooks | Dwolla Connect | Dwolla API Reference
+  description: Endpoints for retrieving, retrying, and listing retries for webhooks.
+---
+
+import Webhook from "../../../../assets/api-resource-tables/webhooks/Webhook";
+
+# Webhooks
+
+When a new event is created, and there is an active [webhook subscription](/api-reference/connect/webhook-subscriptions), a new webhook is created in order to deliver that event. Attempted deliveries are recorded under the webhook's `attempts` property. Each attempt includes the recorded request and response of the delivery attempt. Webhooks are sent asynchronously and are not guaranteed to be delivered in order. We recommend that applications protect against duplicated events by [making event processing idempotent](/guides/webhooks/process-validate#check-for-duplicate-events).
```
Comment (spencerhunter) [inline pages/api-reference/connect/webhooks/index.mdx]: Let's remove this link for now
```suggestion
We recommend that applications protect against duplicated events by [making event processing idempotent](/guides/webhooks/process-validate#check-for-duplicate-events).
```

File: pages/api-reference/connect/webhook-subscriptions/list-webhooks.mdx [hunk 8]
```diff
@@ -0,0 +1,115 @@
+---
+layout: guides
+product: connect
+title: List Webhooks
```
Comment (spencerhunter) [inline pages/api-reference/connect/webhook-subscriptions/list-webhooks.mdx]: should we change this to `List webhooks for a webhook subscription`?

**Review by spencerhunter — CHANGES_REQUESTED** [review]
@ShreyaThapa Added a few comments! Overall, it looks good. Update endpoint titles to be sentence case and it should be g2g

**Comment by ShreyaThapa** [comment]
Addressed comments and accepted suggestions!

**Review by spencerhunter — APPROVED** [review]

**Pull request merged**